x-pack/plugins/stack_alerts/public/alert_types/geo_containment/readme.md

```diff
@@ -0,0 +1,160 @@
+project. To edit it, open this file in your editor of choice, add the line described in
+the next step to the bottom of the file (or really anywhere) and save. For more details
+on different config modifications or on how to make production config modifications,
+see [the current docs](https://www.elastic.co/guide/en/kibana/current/settings.html)
+- Set the following configuration settings in your `config/kibana.yml`:
+`xpack.stack_alerts.enableGeoAlerting: true`
+
+### 2. Run ES/Kibana dev env with ssl enabled
+- In two terminals, run the normal commands to launch both elasticsearch and kibana but 
+append `--ssl` to the end of each as an arg, i.e.:
+  - `yarn es snapshot --ssl  # Runs Elasticsearch`
+  - `yarn start --ssl # Runs Kibana`
+  
+### 3. Get an MTA data api key
+- You'll need to obtain an NYC MTA api key, you can request this
+  key [here](https://docs.google.com/forms/d/e/1FAIpQLSfGUZA6h4eHd2-ImaK5Q_I5Gb7C3UEP5vYDALyGd7r3h08YKg/viewform?hl=en&formkey=dG9kcGIxRFpSS0NhQWM4UjA0V0VkNGc6MQ#gid=0)
+
+### 4. Get trackable point data (MTA bus data) into elasticsearch
+- You'll be using the script: `https://github.com/thomasneirynck/mtatracks` to harvest
+live bus data to populate the system. Clone the repo and follow the instructions in
+the readme to set up. 
+- Using the MTA key you obtained in the previous step, the final command to run
+in a local terminal should look something like the following. This script loads large
+quantities of data the frequency listed below (20000ms = 20s) or higher:
+`node ./load_tracks.js -a <YOUR_API_KEY> -f 20000`
+
+### 5. Open required Kibana tabs
+There are 4 separate tabs you'll need for a combination of loading and viewing the
+data. Since you'll be jumping between them, it might be easiest to just open them
+upfront. Each is preceded by `https://localhost:5601/<your dev env prefix>/app/`:
+- Stack Management > Index Patterns: `management/kibana/indexPatterns`
+- Stack Management > Alerts & Actions: `management/insightsAndAlerting/triggersActions/alerts`
+- Dev tools: `dev_tools#/console`
+- Maps: `maps`
+
+### 6. Create a new alert index in dev tools
+Execute the following two commands:
+- Create index
+```
+PUT /manhattan_mta_alerts
+```
+- Create mappings
+```
+PUT /manhattan_mta_alerts/_mapping
+{
+  "properties": {
+    "entityId": {
+      "type": "keyword"
+    },
+    "entityDateTime": {
+      "type": "date"
+    },
+    "entityDocumentId": {
+      "type": "keyword"
+    },
+    "detectionDateTime": {
+      "type": "date"
+    },    
+    "entityLocation": {
+      "type": "geo_point"
+    },
+    "containingBoundaryId": {
+      "type": "keyword"
+    },
+    "containingBoundaryName": {
+      "type": "keyword"
+    }
+  }
+}
+```
+
+### 7 Create map to monitor alerts
+- Go to the Maps app
+- Create a new map
+- Using GeoJSON Upload, upload the geojson file located in the folder of the previously 
+cloned `mta_tracks` repo: `nyc-neighborhoods.geo.json`. Accept all of the default
```

Comment:
iso. having this feel synthetic wrt nyc boundaries: what about a geojson of something more "real world". Like for example, Times Square or perimeter around UN, and just check if vehicles enter/leave those areas.

Reply:
This was actually the only one I held off on since it's more related to the file contained in https://github.com/thomasneirynck/mtatracks. What we might want to do is throw up a separate PR against that repo, get something together that we agree on, then do a small PR to update the reference in this one. If we're going to craft something a little nicer, it would also be worth rethinking the tracks so that they hit the major NYC sites in the GeoJSON file

Reply:
FYI: https://github.com/thomasneirynck/faketracks/pull/2

Reply:
thx, merged!